src/services/GraphQLService.ts

```diff
@@ -1,48 +1,233 @@
-   * @param variables optional: additional variables for the query, if variables were used
-   * @returns a promise that resolves to the query's results, or rejects if the query failed
-   * @throws when the 'query' argument is undefined or an empty string
-   */
-  static query(query: string, variables?: { [key: string]: any }): Promise<GraphqlResponse> {
-    if (!query || query.length === 0) {
-      throw new Error('GraphQLService query expects a GraphQL query as its first argument');
-    }
-    return ApiBase.queryGraph(query.replace(/\s\s+/g, ' ').trim(), variables);
-  }
+class GraphQLService {
+
+    private apiKey: string;
+
+    private apiServer: string;
+
+    private socket: WebSocket = null;
+
+    private connectionStatus = ConnectionStatus.DISCONNECTED;
+
+    private nextSubscriptionId: number = 1;
+
+    private subscriptions: Subscription[] = [];
+
+    private subscriptionHandlerMap: { [id: string]: Function } = {};
+
+    /** A timeout object after which to retry connecting to Qminder API. */
+    private retryTimeout: any;
+
+    /** Counts the amount of times the event emitter retried connecting. This is used for
+     *  exponential retry falloff. */
+    private connectionRetries = 0;
+
+    /**
+     * Query Qminder API with GraphQL.
+     *
+     * Send a GraphQL query to the Qminder API.
+     *
+     * When the query contains variables, make sure to fill them all in the second parameter.
+     *
+     * For example:
+     *
+     * ```javascript
+     * import * as Qminder from 'qminder-api';
+     * Qminder.setKey('API_KEY_HERE');
+     * // 1. Figure out the selected location ID of the current user, with async/await
+     * try {
+     *     const response = await Qminder.graphql.query(`{ me { selectedLocation } }`);
+     *     console.log(response.me.selectedLocation); // "12345"
+     * } catch (error) {
+     *     console.log(error);
+     * }
+     * // 2. Figure out the selected location ID of the current user, with promises
+     * Qminder.graphql.query("{ me { selectedLocation } }").then(function(response) {
+     *     console.log(response.me.selectedLocation);
+     * }, function(error) {
+     *     console.log(error);
+     * });
+     * ```
+     *
+     * @param query required: the query to send, for example `"{ me { selectedLocation } }"`
+     * @param variables optional: additional variables for the query, if variables were used
+     * @returns a promise that resolves to the query's results, or rejects if the query failed
+     * @throws when the 'query' argument is undefined or an empty string
+     */
+    query(query: string, variables?: { [key: string]: any }): Promise<GraphqlResponse> {
+        if (!query || query.length === 0) {
+            throw new Error('GraphQLService query expects a GraphQL query as its first argument');
+        }
+        return ApiBase.queryGraph(query.replace(/\s\s+/g, ' ').trim(), variables);
+    }
+
+    subscribe(query: string): Observable<object> {
```

Comment:
Please write a docstring with examples, param descriptions etc here. 

This is needed for: 

1) IDE completions: 

![image](https://user-images.githubusercontent.com/358714/49457537-b45e6980-f7f3-11e8-928f-ede48f8da5e3.png)

2) Autogenerated documentation (which is a bit broken right now - but this doesn't excuse the tech debt)